pkg/scheduler/framework/plugins/podtopologyspread/scoring_test.go

```diff
@@ -147,6 +147,41 @@ func TestPreScoreStateEmptyNodes(t *testing.T) {
 				TopologyNormalizingWeight: []float64{topologyNormalizingWeight(1), topologyNormalizingWeight(1)},
 			},
 		},
+		{
+			name: "system defaults constraints and nodes don't have label zone",
+			pod:  st.MakePod().Name("p").Namespace("default").Label("foo", "tar").Label("baz", "sup").OwnerReference("rs1", appsv1.SchemeGroupVersion.WithKind("ReplicaSet")).Obj(),
+			config: config.PodTopologySpreadArgs{
+				DefaultingType: config.SystemDefaulting,
+			},
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-a").Label(v1.LabelHostname, "node-a").Label(v1.LabelTopologyZone, "zone1").Obj(),
+				st.MakeNode().Name("node-x").Label(v1.LabelHostname, "node-x").Obj(),
+				st.MakeNode().Name("node-y").Label(v1.LabelHostname, "node-y").Obj(),
+			},
+			objs: []runtime.Object{
+				&appsv1.ReplicaSet{ObjectMeta: metav1.ObjectMeta{Namespace: "default", Name: "rs1"}, Spec: appsv1.ReplicaSetSpec{Selector: st.MakeLabelSelector().Exists("foo").Obj()}},
+			},
+			want: &preScoreState{
+				Constraints: []topologySpreadConstraint{
+					{
+						MaxSkew:     3,
+						TopologyKey: v1.LabelHostname,
+						Selector:    mustConvertLabelSelectorAsSelector(t, st.MakeLabelSelector().Exists("foo").Obj()),
+					},
+					{
+						MaxSkew:     5,
+						TopologyKey: v1.LabelTopologyZone,
+						Selector:    mustConvertLabelSelectorAsSelector(t, st.MakeLabelSelector().Exists("foo").Obj()),
+					},
+				},
+				IgnoredNodes: sets.NewString(),
+				TopologyPairToPodCounts: map[topologyPair]*int64{
+					{key: v1.LabelTopologyZone, value: "zone1"}: pointer.Int64Ptr(0),
+					{key: v1.LabelTopologyZone, value: ""}:      pointer.Int64Ptr(0),
+				},
+				TopologyNormalizingWeight: []float64{topologyNormalizingWeight(3), topologyNormalizingWeight(2)},
+			},
+		},
```

Comment:
Can you add a case to cover that DefaultingType is "ListDefaulting", and the nodes don't have the required topology key specified in the user-defined constraints?

Reply:
done